ably/rest_client_test.go

```diff
@@ -567,16 +577,21 @@ func TestFixConnLeak_ISSUE89(t *testing.T) {
 		}
 	}
 
-	for _, v := range trackRecord {
-		v.Conn.SetReadDeadline(time.Now())
-		if _, err := v.Conn.Read(make([]byte, 1)); err != nil {
-			if !connIsClosed(err) {
-				t.Errorf("expected conn %s to be closed", v.Conn.LocalAddr())
-			}
+	for _, c := range conns {
+		if !ablytest.Before(1 * time.Second).IsTrue(func() bool {
+			return atomic.LoadUintptr(&c.closed) != 0
+		}) {
+			t.Errorf("conn to %v wasn't closed", c.RemoteAddr())
 		}
 	}
 }
 
-func connIsClosed(err error) bool {
-	return strings.Contains(err.Error(), "use of closed network connection")
+type connTrackingClose struct {
```

Comment:
Maybe we can rename it to `connCloseTracking` / `connCloseTracker`

Reply:
Done at 506ab43.